Objects/unionobject.c

```diff
@@ -108,25 +108,282 @@ static PyMethodDef union_methods[] = {
 static PyObject *
 union_richcompare(PyObject *a, PyObject *b, int op)
 {
+    PyObject* b_set = NULL;
```

Comment:
This probably should start by testing the value of `op`, and returning `NotImplemented` if it's not `EQ` or `NE`. Also for `NE` it should probably call itself with `op=EQ` and invert the result, or something. (Compare dict_richcompare() in dictobject.c.)